internal/internal_activity.go

```diff
@@ -227,8 +227,8 @@ func validateRetryPolicy(p *shared.RetryPolicy) error {
 	if p.GetMaximumAttempts() < 0 {
 		return errors.New("negative MaximumAttempts on retry policy is invalid")
 	}
-	if p.GetExpirationIntervalInSeconds() <= 0 {
-		return errors.New("ExpirationIntervalInSeconds on retry policy must be greater than 0")
+	if p.GetExpirationIntervalInSeconds() < 0 {
```

Comment:
should add check to make sure max attempt and expiration are not 0 at same time.

Reply:
It is already there on line 236